protoc-gen-openapiv2/internal/genopenapi/helpers.go

```diff
@@ -13,3 +13,25 @@ import (
 func fieldName(k string) string {
 	return strings.ReplaceAll(cases.Title(language.AmericanEnglish).String(k), "-", "_")
 }
+
+//this method will filter the same fields and return the unique one
+func searchRequired(schemaFieldsRequired []string, fieldsRequired []string) []string {
+	var unique []string
+	var index *int
+
+	for _, fieldRequired := range fieldsRequired {
+		index = nil
+		for i, schemaFieldRequired := range schemaFieldsRequired {
+			if schemaFieldRequired == fieldRequired {
+				index = &i
+				break
+			}
+		}
+		if index != nil {
+			unique = append(schemaFieldsRequired[:*index], schemaFieldsRequired[*index+1:]...)
+		} else {
+			unique = schemaFieldsRequired
+		}
+	}
+	return unique
+}
```

Comment:
I'm confused by what this function is supposed to do. Could you explain?

Reply:
so it is comparing the schemfield required slice element to fieldshema required slice element so the if you look at issue #2635 when same field has annotation required and it is also in json schema required array that creates the problem in the issue that is saying few fields are more than once in the required list of fields in openapiv2 doc

Reply:
I see. Could you add some tests for this function please?